src/imars3d/backend/dataio/data.py

```diff
@@ -42,6 +42,32 @@
 # }
 
 
+class Foldernames(param.Foldername):
+    r"""
+    Parameter that can be set to a string specifying the path of a folder, or a list of such strings.
+
+    The string(s) should be specified in UNIX style, but they will be
+    returned in the format of the user's operating system.
+
+    The specified path(s) can be absolute, or relative to either:
+
+    * any of the paths specified in the search_paths attribute (if search_paths is not None);
+
+    or
+
+    * any of the paths searched by resolve_dir_path() (if search_paths is None).
+    """
+
+    def _resolve(self, paths):
+        if isinstance(paths, (str, Path)):
+            return super()._resolve(paths)
+        elif isinstance(paths, (List, Tuple)):
+            return [self._resolve(path) for path in paths]
+        else:
+            name = next(x for x in [self.name, self.label, "Foldernames parameter"] if x)
+            raise ValueError(f"{name} must be a string or a list of strings")
+
+
```

Comment:
I would recommend contributing this back to `param` lib since this is a critical feature they are missing.